tests/model_fields/test_manytomanyfield.py

```diff
@@ -105,3 +106,29 @@ def test_value_from_object_instance_with_pk(self):
         self.assertEqual(
             obj._meta.get_field("m2m").value_from_object(obj), [related_obj]
         )
+
+
+@isolate_apps("model_fields")
+class M2MCompositePKCheckTests(SimpleTestCase):
+    def test_many_to_many_from_model_with_composite_primary_key(self):
```

Comment:
It might be nice to keep this test adjacent to the other test for the other direction `tests.invalid_models_tests.test_relative_fields_test_many_to_many_to_model_with_composite_primary_key`